### ISSUE TITLE
CrossEncropyLoss() Bug: It can not detect wrong label id on gpu side

### ISSUE THREAD

**Issue by wangtingwei1993** [pytorch/pytorch#40858]
## 🐛 Bug

CrossEntropyLoss() Bug: CrossEntropyLoss() can not detect wrong label id on gpu side

## To Reproduce
```
import torch
criterion = torch.nn.CrossEntropyLoss()

# Setup dummy data
bsize = 16
num_classes = 5
# Dummy groundtruth labels
labels = torch.randint(num_classes, (bsize,)).long()
# Dummy network class probability predictions
probs = torch.rand(bsize, num_classes)

# Computing the loss here works correctly
loss = criterion(probs, labels)

#---------
# Expected behavior on the CPU
#---------
# Error case, where a target class id is greater than probs.shape[0]
labels[0] = num_classes + 1
loss = criterion(probs, labels)
# The previous line correctly threw a Python error 'RuntimeError: Assertion \
# `cur_target >= 0 && cur_target < n_classes' failed. '

#---------
# Unexpected behavior on the GPU
#---------
# But when we move to the GPU bad things happen
xpu = torch.device(0)
probs = probs.to(xpu)
labels = labels.to(xpu)

loss = criterion(probs, labels)
# The previous line raises nothing. There should be an error message like on CPU side.

## Expected behavior

There should be a runtime error when the label id is larger than the output dimension (probs.size(1))
```

## Environment

 - PyTorch Version : 1.5.0
 - OS : Linux
 - How you installed PyTorch : Conda
 - Python version: 3.7.6
 - CUDA/cuDNN version: 10.2

cc @ngimel

**Comment by albanD** [comment]
@ngimel is this missing assert in the GPU code or is it expected?

**Comment by ngimel** [comment]
This is fixed in 1.5.1, nightlies and master. Closing, please reopen if the problem persists.
